crates/sequencer/src/sequencer.rs

```diff
@@ -481,6 +474,7 @@ where
 
                 let (soft_confirmation_receipt, checkpoint) = self.stf.end_soft_confirmation(
                     active_fork_spec,
+                    self.state_root.clone().as_ref().to_vec(),
```

Comment:
```suggestion
                    self.state_root.to_vec(),
```

This should work because state_root is &[u8]?